api/graphql/schema.graphql

```diff
@@ -1858,14 +1858,15 @@ type Mutation {
 
   # UserSkill mutations
   createUserSkill(
-    userId: ID! @rename(attribute: "user_id")
-    skillId: ID! @rename(attribute: "skill_id")
+    userId: UUID! @rename(attribute: "user_id")
+    skillId: UUID! @rename(attribute: "skill_id")
```

Comment:
Is there a difference between the two, because it seems both are being used and I can't tell if there is a reason one in used in an instance as opposed to the other. `ID` verse `UUID` 

Below for instances of both being used

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/f624b55b-42fb-41de-b7a5-af4b343db4b5)

Reply:
Yeah, `UUID` ensures it is an actual UUID and `ID` is just `string`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/d989f6c340f69676824e275835f40cf545a301ba/api/app/GraphQL/Scalars/UUID.php#L66-L75

Reply:
Should we then enforce usage of `UUID` over `ID` then?

Reply:
Uh-oh 😆 
Can we remove ID as a type so as to prevent that from happening again? 🤔

Reply:
`ID` is still a valid type we could use, just not when we are referring to a `UUID`. So, I think we should keep it around.